### PR TITLE
[FEATURE] Refactor `maps.py`

### LINKED ISSUE #98: [FEATURE] Refactor `maps.py`
### Feature Description

The current `maps.py` file is ~1400 LOC. To improve maintainability, it will be good to separate it out into distinct modules.

### Implementation (Optional)

Logically distinct modules like - `maps/gates.py`, `maps/constants.py` and `maps/expressions.py`

### PR DESCRIPTION
**Pull request by devilkiller-ag**
Resolves #98 

## Summary of changes

This PR refactors `maps.py` to segregate gates and expressions related to functions and constants. It replaces the `maps.py` file with two new files, `gates.py` and `expressions.py`, stored in the `maps` folder. The segregation done is as follows:

`/src/pyqasm/map/__init__.py`:
- OperatorFunction
- SWITCH_BLACKLIST_STMTS
- SUPPORTED_QASM_VERSIONS
- QUANTUM_STATEMENTS

`/src/pyqasm/map/gates.py`:
- u3_gate
- u3_inv_gate
- u2_gate
- u2_inv_gate
- global_phase_gate
- sxdg_gate_op
- cy_gate
- ch_gate
- xy_gate
- xx_plus_yy_gate
- ryy_gate
- zz_gate
- phaseshift_gate
- cswap_gate
- pswap_gate
- iswap_gate
- crx_gate
- cry_gate
- crz_gate
- cu_gate
- cu3_gate
- cu1_gate
- csx_gate
- rxx_gate
- rccx_gate
- rzz_gate
- cphaseshift_gate
- cphaseshift00_gate
- cphaseshift01_gate
- cphaseshift10_gate
- gpi_gate
- gpi2_gate
- ms_gate
- ccx_gate_op
- ecr_gate
- c3sx_gate
- c4x_gate
- prx_gate
- one_qubit_gate_op
- one_qubit_rotation_op
- two_qubit_gate_op
- map_qasm_op_to_callable
- map_qasm_inv_op_to_callable

`/src/pyqasm/map/constants.py`:
- OPERATOR_MAP
- ONE_QUBIT_OP_MAP
- ONE_QUBIT_ROTATION_MAP
- TWO_QUBIT_OP_MAP
- THREE_QUBIT_OP_MAP
- FOUR_QUBIT_OP_MAP
- FIVE_QUBIT_OP_MAP
- SELF_INVERTING_ONE_QUBIT_OP_SET
- ST_GATE_INV_MAP
- ROTATION_INVERSION_ONE_QUBIT_OP_MAP
- U_INV_ROTATION_MAP

`/src/pyqasm/map/expressions.py`:
- LIMITS_MAP
- CONSTANTS_MAP
- VARIABLE_TYPE_MAP
- VARIABLE_TYPE_CAST_MAP
- ARRAY_TYPE_MAP
- MAX_ARRAY_DIMENSIONS
- qasm3_expression_op_map
- qasm_variable_type_cast

### CODE REVIEW

File: src/pyqasm/expressions.py [hunk 1]
```diff
@@ -34,7 +34,8 @@
 
 from pyqasm.analyzer import Qasm3Analyzer
 from pyqasm.exceptions import ValidationError, raise_qasm3_error
-from pyqasm.maps import CONSTANTS_MAP, qasm3_expression_op_map
+from pyqasm.maps.gates import qasm3_expression_op_map
```
Comment (TheGupta2012) [inline src/pyqasm/expressions.py:37]: `qasm3_expression_op_map` should be present in `pyqasm.maps.expressions.py`

File: src/pyqasm/maps/__init__.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    QuantumGate,
+    QuantumGateDefinition,
+    QuantumMeasurementStatement,
+    QuantumReset,
+    QubitDeclaration,
+    SubroutineDefinition,
+)
+
+# Define the type for the operator functions
+OperatorFunction = Union[
```
Comment (TheGupta2012) [inline src/pyqasm/maps/__init__.py:30]: Why do we need `OperatorFunction` in this file? Better to move to `expressions.py` as only that file is using it

File: src/pyqasm/maps/gates.py [hunk 3]
```diff
@@ -1,1335 +1,1221 @@
...
+)
+
+from pyqasm.elements import InversionOp
+from pyqasm.exceptions import ValidationError
+from pyqasm.linalg import kak_decomposition_angles
+from pyqasm.maps import OperatorFunction
+from pyqasm.maps.expressions import CONSTANTS_MAP
+
+
+OPERATOR_MAP: dict[str, OperatorFunction] = {
```
Comment (TheGupta2012) [inline src/pyqasm/maps/gates.py:37]: This should be moved to `expressions.py`

File: src/pyqasm/maps/gates.py [hunk 4]
```diff
@@ -1,1335 +1,1221 @@
...
+    "|": lambda x, y: x | y,
+    "<<": lambda x, y: x << y,
+    ">>": lambda x, y: x >> y,
+    "~": lambda x: ~x,
+    "!": lambda x: not x,
+    "UMINUS": lambda x: -x,
+}
+
+
+def qasm3_expression_op_map(op_name: str, *args) -> Union[float, int, bool]:
```
Comment (TheGupta2012) [inline src/pyqasm/maps/gates.py:62]: similar as above, to be moved to `expressions.py`

**Review by TheGupta2012 — CHANGES_REQUESTED** [review]
Hey, thanks for the changes! I've added some comments but other than that looks good.

File: src/pyqasm/expressions.py (hunk 1 above)
Comment (devilkiller-ag) [reply src/pyqasm/expressions.py]: Done

File: src/pyqasm/maps/__init__.py (hunk 2 above)
Comment (devilkiller-ag) [reply src/pyqasm/maps/__init__.py]: Done

File: src/pyqasm/maps/gates.py (hunk 3 above)
Comment (devilkiller-ag) [reply src/pyqasm/maps/gates.py]: Done

File: src/pyqasm/maps/gates.py (hunk 4 above)
Comment (devilkiller-ag) [reply src/pyqasm/maps/gates.py]: Done

**Review by TheGupta2012 — APPROVED** [review]
Thanks for working on the suggestions! Changes lgtm

**Pull request merged**